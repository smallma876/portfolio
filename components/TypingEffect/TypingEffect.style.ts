import styled from "styled-components";

const Wrapper = styled.div`
  color: #fff;
  font-size: 1.5rem;

  h1 {
    margin: 0;
    height: 4rem;
  }
`;

export const TypingEffectStyle = {
  Wrapper,
};
