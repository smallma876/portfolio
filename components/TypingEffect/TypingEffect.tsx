import React, { ElementType, FC, useEffect, useRef, useState } from "react";
import { TypingEffectStyle } from "./TypingEffect.style";

interface TypingEffectProps {
  component: ElementType;
  variant: "h1" | "h2" | "h3";
  text: string;
}

const TypingEffect: FC<TypingEffectProps> = ({
  component: Component,
  variant,
  text,
}: TypingEffectProps) => {
  const [splitText, setSplitText] = useState("");
  const sizeLetter = useRef(0);

  useEffect(() => {
    /* istanbul ignore next */
    const addLetter = () => {
      let nexLetter = text.substring(
        sizeLetter.current,
        sizeLetter.current + 1
      );
      sizeLetter.current = sizeLetter.current + 1;
      setSplitText((previousState) => {
        return previousState.concat(nexLetter);
      });
    };
    /* istanbul ignore else */
    if (splitText.length < text.length) {
      setTimeout(addLetter, 500);
    } else {
      sizeLetter.current = 0;
      setTimeout(() => setSplitText(""), 500);
    }
  }, [splitText, text]);

  return (
    <TypingEffectStyle.Wrapper>
      <Component>{splitText}</Component>
    </TypingEffectStyle.Wrapper>
  );
};

export default TypingEffect;
