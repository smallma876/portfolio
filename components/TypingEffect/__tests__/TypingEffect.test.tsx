import { render, screen } from "@testing-library/react";
import TypingEffect from "../TypingEffect";

const renderComponent = async () => {
  const { container } = await render(
    <TypingEffect component="h1" variant="h1" text="I'm Sergio" />
  );

  return container;
};

describe("TypingEffect", () => {
  it("should render match snapshot", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
