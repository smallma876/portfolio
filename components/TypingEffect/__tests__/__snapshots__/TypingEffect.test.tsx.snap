// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypingEffect should render match snapshot 1`] = `Promise {}`;
